iothub/device/src/ClientPropertyCollection.cs

```diff
@@ -17,162 +16,35 @@ public class ClientPropertyCollection : PayloadCollection
     {
         private const string VersionName = "$version";
 
-        /// <summary>
-        /// Adds the value to the collection.
-        /// </summary>
-        /// <remarks>
-        /// If the collection already has a key matching a property name supplied this method will throw an <see cref="ArgumentException"/>.
-        /// <para>
-        /// When using this as part of the writable property flow to respond to a writable property update you should pass in the value
-        /// as an instance of <see cref="PayloadSerializer.CreateWritablePropertyResponse(object, int, long, string)"/>
-        /// to ensure the correct formatting is applied when the object is serialized.
-        /// </para>
-        /// </remarks>
-        /// <param name="propertyName">The name of the property to add.</param>
-        /// <param name="propertyValue">The value of the property to add.</param>
-        /// <exception cref="ArgumentNullException"><paramref name="propertyName"/> is <c>null</c>.</exception>
-        /// <exception cref="ArgumentException"><paramref name="propertyName"/> already exists in the collection.</exception>
-        public void AddRootProperty(string propertyName, object propertyValue)
```

Comment:
Remove this overload that throws if the value already exists. Note the `AddOrUpdate` overload is renamed to just `AddRootProperty` and effectively replaces this.